src/main/java/org/motechproject/mots/service/UserService.java

```diff
@@ -116,7 +116,7 @@ public User getUser(UUID id) {
   public User saveUser(User user, boolean encodeNewPassword) {
     if (encodeNewPassword) {
       String newPasswordEncoded = new BCryptPasswordEncoder().encode(user.getPassword());
-      user.setPassword(newPasswordEncoded);
+      user.setPassword("{bcrypt}" + newPasswordEncoded);
```

Comment:
I don't think it's the best solution, you can update the password encoder instead